webapp/channels/src/components/channel_header_menu/channel_header_menu.tsx

```diff
@@ -0,0 +1,183 @@
+import ChannelHeaderTitleGroup from '../channel_header/channel_header_title_group';
+
+type Props = {
+    dmUser?: UserProfile;
+    gmMembers?: UserProfile[];
+    archivedIcon?: JSX.Element;
+    sharedIcon?: JSX.Element;
+    isMobile?: boolean;
+}
+
+export default function ChannelHeaderMenu({dmUser, gmMembers, isMobile, archivedIcon, sharedIcon}: Props): JSX.Element | null {
+    const intl = useIntl();
+
+    const user = useSelector(getCurrentUser);
+    const channel = useSelector(getCurrentChannel);
+    const isDefault = useSelector(isCurrentChannelDefault);
+    const isFavorite = useSelector(isCurrentChannelFavorite);
+    const isMuted = useSelector(isCurrentChannelMuted);
+    const isLicensedForLDAPGroups = useSelector(getLicense).LDAPGroups === 'true';
+    const pluginMenuItems = useSelector(getChannelHeaderMenuPluginComponents);
+
+    const isReadonly = false;
+
+    if (!channel) {
+        return null;
+    }
+
+    const isDirect = (channel.type === Constants.DM_CHANNEL);
+    const isGroup = (channel.type === Constants.GM_CHANNEL);
+
+    let channelTitle: ReactNode = channel.display_name;
+    let ariaLabel = intl.formatMessage({
+        id: 'channel_header.otherchannel',
+        defaultMessage: '{displayName} Channel Menu',
+    }, {
+        displayName: channel.display_name,
+    });
+    if (isDirect && dmUser) {
+        channelTitle = <ChannelHeaderTitleDirect dmUser={dmUser}/>;
+        if (user.id === dmUser.id) {
+            ariaLabel = intl.formatMessage({
+                id: 'channel_header.directchannel',
+                defaultMessage: '{displayName} (you) Channel Menu',
+            }, {
+                displayName: channel.display_name,
+            });
+        }
+    } else if (isGroup) {
+        channelTitle = <ChannelHeaderTitleGroup gmMembers={gmMembers}/>;
+    }
+
+    const pluginItems = pluginMenuItems.map((item) => {
+        const handlePluginItemClick = () => {
+            if (item.action) {
+                item.action(channel.id);
+            }
+        };
+
+        return (
+            <Menu.Item
+                id={item.id + '_pluginmenuitem'}
+                key={item.id + '_pluginmenuitem'}
+                onClick={handlePluginItemClick}
+                labels={<span>{item.text}</span>}
+            />
+        );
+    });
+
+    return (
+        <Menu.Container
+            menuButtonTooltip={{
+                text: channelTitle as string,
+            }}
+            menuButton={{
+                id: 'channelHeaderDropdownButton',
+                class: classNames('channel-header__trigger style--none'),
+                children: (
+                    <>
+                        {archivedIcon}
+                        <strong
+                            id='channelHeaderTitle'
+                            className='heading'
+                            role='heading'
+                            aria-level={2}
```

Comment:
Could you remove these? We actually removed these in https://github.com/mattermost/mattermost/pull/30277
```suggestion
```

Reply:
done